handler/src/test/java/io/netty/handler/pcap/PcapWriteHandlerTest.java

```diff
@@ -0,0 +1,137 @@
+                .handler(new PcapWriteHandler(new ByteBufOutputStream(byteBuf)));
+
+        ChannelFuture channelFutureClient = client.connect(srvAddr, cltAddr).sync();
+        assertTrue(channelFutureClient.isSuccess());
+        assertTrue(channelFutureClient.channel().writeAndFlush(Unpooled.wrappedBuffer("Meow".getBytes()))
+                .sync().isSuccess());
+        assertTrue(eventLoopGroup.shutdownGracefully().sync().isSuccess());
+
+        // Verify Pcap Global Headers
+        assertEquals(0xa1b2c3d4, byteBuf.readInt()); // magic_number
+        assertEquals(2, byteBuf.readShort());        // version_major
+        assertEquals(4, byteBuf.readShort());        // version_minor
+        assertEquals(0, byteBuf.readInt());          // thiszone
+        assertEquals(0, byteBuf.readInt());          // sigfigs
+        assertEquals(0xffff, byteBuf.readInt());     // snaplen
+        assertEquals(1, byteBuf.readInt());          // network
+
+        // Verify Pcap Packet Header
+        byteBuf.readInt(); // Just read, we don't care about timestamps for now
+        byteBuf.readInt(); // Just read, we don't care about timestamps for now
+        assertEquals(46, byteBuf.readInt()); // Length of Packet Saved In Pcap
+        assertEquals(46, byteBuf.readInt()); // Actual Length of Packet
+
+        // -------------------------------------------- Verify Packet --------------------------------------------
+        // Verify Ethernet Packet
+        ByteBuf ethernetPacket = byteBuf.readBytes(46);
+        ByteBuf dstMac = ethernetPacket.readBytes(6);
+        ByteBuf srcMac = ethernetPacket.readBytes(6);
+        assertArrayEquals(new byte[]{0, 0, 94, 0, 83, -1}, ByteBufUtil.getBytes(dstMac));
+        assertArrayEquals(new byte[]{0, 0, 94, 0, 83, 0}, ByteBufUtil.getBytes(srcMac));
+        assertEquals(0x0800, ethernetPacket.readShort());
+
+        // Verify IPv4 Packet
+        ByteBuf ipv4Packet = ethernetPacket.readBytes(32);
+        assertEquals(0x45, ipv4Packet.readByte());    // Version + IHL
+        assertEquals(0x00, ipv4Packet.readByte());    // DSCP
+        assertEquals(32, ipv4Packet.readShort());     // Length
+        assertEquals(0x0000, ipv4Packet.readShort()); // Identification
+        assertEquals(0x0000, ipv4Packet.readShort()); // Fragment
+        assertEquals((byte) 0xff, ipv4Packet.readByte());      // TTL
+        assertEquals((byte) 17, ipv4Packet.readByte());        // Protocol
+        assertEquals(0, ipv4Packet.readShort());      // Checksum
+        // Source IPv4 Address
+        assertEquals(NetUtil.ipv4AddressToInt((Inet4Address) srvAddr.getAddress()), ipv4Packet.readInt());
+        // Destination IPv4 Address
+        assertEquals(NetUtil.ipv4AddressToInt((Inet4Address) cltAddr.getAddress()), ipv4Packet.readInt());
+
+        // Verify UDP Packet
+        ByteBuf udpPacket = ipv4Packet.readBytes(12);
+        assertEquals(cltAddr.getPort() & 0xffff, udpPacket.readShort() & 0xffff); // Source Port
```

Comment:
Maybe instance of `udpPacket.readShort() & 0xffff`  we can use `buf.readUnsignedShort()` ?

Reply:
Yeah, Fixed.